stl/inc/span

```diff
@@ -0,0 +1,778 @@
+                "size of span in bytes exceeds std::numeric_limits<size_t>::max()");
+        }
+#endif // _CONTAINER_DEBUG_LEVEL > 0
+        return size() * sizeof(element_type);
+    }
+
+    _NODISCARD constexpr bool empty() const noexcept {
+        return size() == 0;
+    }
+
+    // [span.elem] Element access
+    _NODISCARD constexpr reference operator[](const size_type _Off) const noexcept /* strengthened */ {
+#if _CONTAINER_DEBUG_LEVEL > 0
+        _STL_VERIFY(_Off < size(), "span index out of range");
+#endif // _CONTAINER_DEBUG_LEVEL > 0
+        return _Mydata[_Off];
+    }
+
+    _NODISCARD constexpr reference front() const noexcept /* strengthened */ {
+        if constexpr (_Extent != dynamic_extent) {
+            static_assert(_Extent > 0, "front of empty span");
+        }
+#if _CONTAINER_DEBUG_LEVEL > 0
+        else {
+            _STL_VERIFY(size() > 0, "front of empty span");
+        }
+#endif // _CONTAINER_DEBUG_LEVEL > 0
+        return _Mydata[0];
+    }
+
+    _NODISCARD constexpr reference back() const noexcept /* strengthened */ {
+        if constexpr (_Extent != dynamic_extent) {
+            static_assert(_Extent > 0, "back of empty span");
+        }
+#if _CONTAINER_DEBUG_LEVEL > 0
+        else {
+            _STL_VERIFY(size() > 0, "back of empty span");
+        }
+#endif // _CONTAINER_DEBUG_LEVEL > 0
+        return _Mydata[size() - 1];
+    }
+
+    _NODISCARD constexpr pointer data() const noexcept {
+        return _Mydata;
+    }
+
+    // [span.iterators] Iterator support
+    _NODISCARD constexpr iterator begin() noexcept {
+#if _ITERATOR_DEBUG_LEVEL >= 1
+        return {_Mydata, _Mydata, _Mydata + size()};
+#else // ^^^ _ITERATOR_DEBUG_LEVEL >= 1 ^^^ // vvv _ITERATOR_DEBUG_LEVEL == 0 vvv
+        return {_Mydata};
+#endif // _ITERATOR_DEBUG_LEVEL
+    }
+
+    _NODISCARD constexpr iterator end() noexcept {
+#if _ITERATOR_DEBUG_LEVEL >= 1
+        return {_Mydata + size(), _Mydata, _Mydata + size()};
+#else // ^^^ _ITERATOR_DEBUG_LEVEL >= 1 ^^^ // vvv _ITERATOR_DEBUG_LEVEL == 0 vvv
+        return {_Mydata + size()};
+#endif // _ITERATOR_DEBUG_LEVEL
+    }
+
+    _NODISCARD constexpr const_iterator begin() const noexcept {
+#if _ITERATOR_DEBUG_LEVEL >= 1
+        return {_Mydata, _Mydata, _Mydata + size()};
+#else // ^^^ _ITERATOR_DEBUG_LEVEL >= 1 ^^^ // vvv _ITERATOR_DEBUG_LEVEL == 0 vvv
+        return {_Mydata};
+#endif // _ITERATOR_DEBUG_LEVEL
+    }
+
+    _NODISCARD constexpr const_iterator end() const noexcept {
+#if _ITERATOR_DEBUG_LEVEL >= 1
+        return {_Mydata + size(), _Mydata, _Mydata + size()};
+#else // ^^^ _ITERATOR_DEBUG_LEVEL >= 1 ^^^ // vvv _ITERATOR_DEBUG_LEVEL == 0 vvv
+        return {_Mydata + size()};
+#endif // _ITERATOR_DEBUG_LEVEL
+    }
+
+    _NODISCARD constexpr const_iterator cbegin() const noexcept {
+        return begin();
+    }
+
+    _NODISCARD constexpr const_iterator cend() const noexcept {
+        return end();
+    }

```

Comment:
Similar comment about debug performance - these are currently calling the begin/end member functions, and then calling the iterator conversion operator. I think cbegin/cend are important enough to micro-optimize, by copy-pasting the begin/end implementations above. (But the `reverse_iterator` functions are not important, so they should be left as-is.)

Reply:
Done